sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -840,8 +840,8 @@ Mono<Response<Void>> deleteEntityWithResponse(String partitionKey, String rowKey
         context = TableUtils.setContext(context);
         eTag = ifUnchanged ? eTag : "*";
 
-        if (isNullOrEmpty(partitionKey) || isNullOrEmpty(rowKey)) {
-            return monoError(logger, new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null."));
+        if (partitionKey == null || rowKey == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'partitionKey' and 'rowKey' cannot be null"));
```

Comment:
NIT: is `logger` should be a static `LOGGER`

Reply:
@mssfang What's the benefit of using a static LOGGER over a logger? It's always been non-static so I haven't questioned it. We can look into that in another PR.

Reply:
My bad. wrong suggestion. If the class is a static class then we accept having a shared logger instance